### ISSUE TITLE
lint .value from StrEnum

### ISSUE THREAD

**Issue by asukaminato0721** [astral-sh/ruff#20808]
### Summary

```py
class A(StrEnum):
    AA = auto()
    BB = auto()

A.AA.value # this trigger a lint
A.AA # this is good
```

**Comment by MichaReiser** [comment]
@amyreese what do you think of such a rule?

**Comment by sharkdp** [comment]
@asukaminato0721 Can you clarify why you would want `A.AA.value` to be flagged? As far as I can tell, there is nothing wrong with accessing `.value` on `StrEnum` (it's equivalent to `"aa"` for this example using `auto()`).

Do you want to avoid confusion between the actual enum member (`A.AA`) and that string value (`A.AA.value == "aa"`)? I think there are valid use cases for accessing `.value` on an enum member, so it's unclear to me how you would avoid false positives.

**Comment by asukaminato0721** [comment]
I am migrating from `Enum` to `StrEnum`, then I find use `A.AA` is shorter, and users don't need to know the `.value` behind it. So I think unifying them is a method.

**Comment by amyreese** [comment]
I think this could be a reasonable rule. With `StrEnum` specifically, the enum element itself is intended to be used in place of `.value`, and has all/most of the properties of a normal string:

```pycon
>>> from enum import StrEnum, auto
>>> class Foo(StrEnum):
...     alpha = auto()
...     beta = auto()
...     gamma = auto()
...     delta = auto()
...
>>> Foo.alpha
<Foo.alpha: 'alpha'>
>>> str(Foo.alpha)
'alpha'
>>> Foo.alpha == "alpha"
True
>>> Foo.alpha.startswith("a")
True
>>> sorted(Foo)
[<Foo.alpha: 'alpha'>, <Foo.beta: 'beta'>, <Foo.delta: 'delta'>, <Foo.gamma: 'gamma'>]
```
